frontend/src/metabase/nav/components/DatabasePromptBanner/DatabasePromptBanner.tsx

```diff
@@ -0,0 +1,59 @@
+import { t } from "ttag";
+import { withRouter } from "react-router";
+import type { Location } from "history";
+
+import Link from "metabase/core/components/Link/Link";
+import { useDatabaseListQuery } from "metabase/common/hooks";
+import { useSelector } from "metabase/lib/redux";
+import { getUserIsAdmin } from "metabase/selectors/user";
+import { getIsPaidPlan } from "metabase/selectors/settings";
+
+import {
+  ConnectDatabaseButton,
+  CallToActions,
+  DatabasePromptBannerRoot,
+  Prompt,
+  GetHelpButton,
+} from "./DatabasePromptBanner.styled";
+
+interface DatabasePromptBannerProps {
+  location: Location;
+}
+
+export const DatabasePromptBanner = withRouter(
```

Comment:
HoCs are discouraged, could you pass `Location` from the nearest component rendered by the router instead?